tests/python/pants_test/engine/legacy/test_build_ignore_integration.py

```diff
@@ -62,7 +62,7 @@ def output_to_list(output_filename):
 
   def test_build_ignore_dependency(self):
     run_result = self.run_pants(['-q',
-                                 'list',
+                                 'dependencies',
```

Comment:
> The list goal used to be usable in order to loosely validate that your BUILD files are correct (and implicitly validating the build graph). Is there some replacement we're offering, with stability guarantees, for "run this goal to error if your build graph is probably significantly incorrect"?

Yea, open question. `./pants test` is the closest answer, but folks usually only run that on a subset of their repository as recommended in https://www.pantsbuild.org/orgs.html#continuous-integration

> That you need to make this change suggests that --build-ignore is a flag whose semantics we're changing from v1 to v2 - this test no longer passes with list because v2 BUILD file parsing ignores this global flag.

No: this is due to the first thing you mentioned. This test is confirming that if a dependency is filtered out, we see an error. But as you pointed out: `list` is no longer walking into dependencies, and so we won't trigger that error. `--build-ignore` and `--pants-ignore` have been in place since we ported BUILD file parsing to v2 (2017?).

Reply:
> > That you need to make this change suggests that --build-ignore is a flag whose semantics we're changing from v1 to v2 - this test no longer passes with list because v2 BUILD file parsing ignores this global flag.
> 
> No: this is due to the first thing you mentioned. This test is confirming that if a dependency is filtered out, we see an error. But as you pointed out: `list` is no longer walking into dependencies, and so we won't trigger that error. `--build-ignore` and `--pants-ignore` have been in place since we ported BUILD file parsing to v2 (2017?).

But this _is_ a semantics change, right? A case which used to error and no longer does... This feels like a much more important thing to have a deprecation in place for (and warn people is changing), than the cache setup thing... This is a behaviour people may actually be relying on (as a basic check of all diffs, to make sure dependencies exist and are valid).

Reply:
Yep, agreed. I was just pointing out that there are not two issues: just the one issue.

Reply:
Following up on this conversation, I think that tackling #7581, and then adding a noisy deprecated option here to strongly encourage folks to read https://www.pantsbuild.org/orgs.html#continuous-integration is going to be the path forward for now.